pkg/pool-utils/contracts/protocol-fees/ProtocolFeeCache.sol

```diff
@@ -96,19 +131,21 @@ abstract contract ProtocolFeeCache is RecoveryMode {
         // solhint-disable-previous-line no-empty-blocks
     }
 
-    function _updateProtocolFeeCache(IProtocolFeePercentagesProvider protocolFeeProvider, uint256 feeType) private {
-        uint256 currentValue = protocolFeeProvider.getFeeTypePercentage(feeType);
+    function _updateSwapProtocolFeeCache(IProtocolFeePercentagesProvider protocolFeeProvider, uint256 feeId) private {
+        uint256 swapFee = protocolFeeProvider.getFeeTypePercentage(feeId);
```

Comment:
I'd either:
 a) inline this entire function
 b) hardcode `_swapFeeId` here instead of receiving it as an argument

If you want to inline, you can change this slightly by doing a single storage write with

```
_cache = FeeTypeCache({ swapFee: swapFee; yieldFee: yieldFee; aumFee: aumFee; });
```

Reply:
Unfortunately hardcoding is not possible because the ID is immutable and we are calling this in the constructor.
We can merge the three updates in just one, but we'd still need to pass the IDs as arguments.

Reply:
You can inline the function into both constrcutor and runtime. I'd measure bytecode size with both approaches, and if inlining is much better then go for it.